File: slack-reader/src/main/java/dev/number6/slackreader/dagger/ConfigurationPortModule.java
package dev.number6.slackreader.dagger;

import dev.number6.slackreader.adaptor.EnvironmentVariableSlackReaderConfigurationAdapter;
import dev.number6.slackreader.port.SlackReaderConfigurationPort;
import dagger.Module;
import dagger.Provides;

@Module
public class ConfigurationPortModule {

    @Provides
    public SlackReaderConfigurationPort providesSlackReaderConfig() {
        return new EnvironmentVariableSlackReaderConfigurationAdapter();
    }
}

File: slack-reader/src/main/java/dev/number6/slackreader/dagger/ConfigurationPortModule.kt
package dev.number6.slackreader.dagger

import dagger.Module
import dagger.Provides
import dev.number6.slackreader.adaptor.EnvironmentVariableSlackReaderConfigurationAdapter

import dev.number6.slackreader.port.SlackReaderConfigurationPort

@Module
class ConfigurationPortModule {
    @Provides
    fun providesSlackReaderConfig(): SlackReaderConfigurationPort {
        return EnvironmentVariableSlackReaderConfigurationAdapter()
    }
}